### PR TITLE
[CH-4323] Add Live Activity Support

### PR DESCRIPTION
**Pull request by froodian**
as per https://docs.google.com/document/d/1ifeLwA5L33Utvg8DJhUIOXwEXUBuKKiWnR-00KQlsuE/edit#heading=h.nfrorqdw1t7c and https://jira.braze.com/browse/CH-4323 which are chocablock full of context

### CODE REVIEW

**Review by christorbraze — APPROVED** [review]

File: notification.go [hunk 1]
```diff
@@ -63,6 +63,14 @@ const (
 	// contact the MDM server. If you set this push type, you must use the topic
 	// from the UID attribute in the subject of your MDM push certificate.
 	PushTypeMDM EPushType = "mdm"
+
+	// PushTypeLiveActivity is used for Live Activities that display various
+	// real-time information. If you set this push type, the topic field must
+	// use your app’s bundle ID with push-type.liveactivity appended to the end.
```
Comment (niccrowley) [inline notification.go]:
```suggestion
	// use your app’s bundle ID with `push-type.liveactivity` appended to the end.
```

**Review by niccrowley — APPROVED** [review]

File: payload/builder.go [hunk 2]
```diff
@@ -218,7 +279,7 @@ func (p *Payload) AlertLaunchImage(image string) *Payload {
 // specifiers in loc-key. See Localized Formatted Strings in Apple
 // documentation for more information.
 //
-//  {"aps":{"alert":{"loc-args":args}}}
+//	{"aps":{"alert":{"loc-args":args}}}
```
Comment (cgiulietti-braze) [inline payload/builder.go:282]: Nit: probably this spacing was not intentional

File: payload/builder_test.go [hunk 3]
```diff
@@ -146,6 +147,45 @@ func TestCategory(t *testing.T) {
...
+	payload := NewPayload().SetDismissalDate(timestamp)
+	b, _ := json.Marshal(payload)
+	assert.Equal(t, `{"aps":{"dismissal-date":1674821640}}`, string(b))
+}
+
+func TestStaleDate(t *testing.T) {
+	timestamp := time.Date(2023, 1, 27, 12, 14, 00, 00, time.UTC).Unix()
+	payload := NewPayload().SetStaleDate(timestamp)
+	b, _ := json.Marshal(payload)
+	assert.Equal(t, `{"aps":{"stale-date":1674821640}}`, string(b))
```
Comment (cgiulietti-braze) [inline payload/builder_test.go:167]: What's 1674821640? should it be in a constant to be more readable?

**Review by cgiulietti-braze — APPROVED** [review]
LGTM just a couple of minor comments

File: payload/builder.go (hunk 2 above)
Comment (froodian) [reply payload/builder.go:282]: it is, it's go turning spaces to tabs, which it is opinionated about

File: payload/builder_test.go (hunk 3 above)
Comment (froodian) [reply payload/builder_test.go:167]: it's the unix timestamp of `time.Date(2023, 1, 27, 12, 14, 00, 00, time.UTC)` - I think it's nice to have the clean string and not need to import fmt

**Review by jonhyman — APPROVED** [review]

**Pull request merged**